include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -218,8 +218,31 @@ class CommonOperatorLoader : public ProtobufLoader {
     } else {
       Tensor *T = getTensorByName(op.input(1));
       auto TH = T->getHandle<size_t>();
+
+      std::vector<int64_t> protoDims(T->size());
+      auto oldDim = in->dims(0);
+      int64_t product = 1;
       for (size_t i = 0, e = T->size(); i != e; i++) {
-        newDim.push_back(TH.raw(i));
+        int64_t new_shape = TH.raw(i);
+
+        if (new_shape == 0) {
+          // shape[i] == 0 means that corresponding element should remain
+          // the same.
+          new_shape = oldDim[i];
+        }
+
+        protoDims[i] = new_shape;
+
+        if (new_shape != -1) {
+          product *= new_shape;
+        }
+      }
+      for (size_t i = 0, e = protoDims.size(); i != e; i++) {
+        if (protoDims[i] == -1)
+          // shape[i] == -1 means that corresponding element should be inferred
+          // from all other elements, so that Tensor size remains the same.
+          protoDims[i] = in->getType(0)->size() / product;
```

Comment:
Same as above -- use `in.getType()` to ensure you're getting the correct output of the `NodeValue`.